src/engine/sparqlExpressions/SparqlExpressionGenerators.h

```diff
@@ -120,10 +119,11 @@ inline auto valueGetterGenerator =
     []<typename ValueGetter, SingleExpressionResult Input>(
         size_t numElements, EvaluationContext* context, Input&& input,
         ValueGetter&& valueGetter)
-    -> cppcoro::generator<std::invoke_result_t<
-        ValueGetter,
-        typename decltype(makeGenerator(Input{}, 0, nullptr))::value_type,
-        EvaluationContext*>> {
+    -> cppcoro::generator<std::invoke_result_t<ValueGetter,
+                                               typename decltype(makeGenerator(
+                                                   std::declval<Input>(), 0,
+                                                   nullptr))::value_type,
+                                               EvaluationContext*>> {
```

Comment:
```suggestion
    -> cppcoro::generator<std::invoke_result_t<ValueGetter,
                                               typename decltype(makeGenerator(
                                                   input, 0,
                                                   nullptr))::value_type,
                                               EvaluationContext*>> {
```
Should work here in a trailing return statement.

Reply:
It does not work. What I gather from the compile error is that it has to with `input` being passed through `std::forward` in `makeGenerator`. This tries to call the `VectorWithMemoryLimit(const VectorWithMemoryLimit&)` copy constructor which is private.

Reply:
then replacing `input` by `AD_FWD(input)` should do the trick, that's what actually done:)
(in `util/Forward.h`, equivalenty to `std::forward<Input>(input)`